tests/unit/test_main.py

```diff
@@ -405,6 +408,89 @@ def test_check_switch_consistency_add(self, *args):
         self.napp.check_switch_consistency(switch)
         mock_install_flows.assert_called()
 
+    @patch("napps.kytos.flow_manager.storehouse.StoreHouse.save_flow")
+    def test_add_overlapping_flow(self, *args):
```

Comment:
@viniarck if you think it is useful, I would suggest adding another unit test for the example I've provided in the other comment, so we can validate the strict/non-strict matching.

Reply:
for sure, @italovalcy, I've added this test case on this commit https://github.com/kytos-ng/flow_manager/pull/44/commits/59b155f7d9c87e3028b777aa38dcc6d28164cc42 based on the example provided, I slightly changed it though to start with 4 flows installed, but same semantics, check it out, and let me know what you think and if you could do a second pass review, thanks. 

I also took the chance to run it on mininet the same example you gave to double check, it worked: 

```
{
    "flow_persistence": {
        "00:00:00:00:00:00:00:01": {
            "0": [
                {
                    "created_at": "2021-12-02T15:04:17",
                    "flow": {
                        "actions": [
                            {
                                "action_type": "output",
                                "port": 4294967293
                            }
                        ],
                        "match": {
                            "dl_type": 35020,
                            "dl_vlan": 3799
                        },
                        "priority": 1000,
                        "table_id": 0
                    }
                },
                {
                    "created_at": "2021-12-02T15:14:16",
                    "flow": {
                        "actions": [
                            {
                                "action_type": "output",
                                "port": 2
                            }
                        ],
                        "match": {
                            "dl_vlan": 100,
                            "in_port": 1
                        },
                        "priority": 10
                    }
                },
                {
                    "created_at": "2021-12-02T15:14:16",
                    "flow": {
                        "actions": [
                            {
                                "action_type": "output",
                                "port": 3
                            }
                        ],
                        "match": {
                            "dl_vlan": 200,
                            "in_port": 1
                        },
                        "priority": 10
                    }
                },
                {
                    "created_at": "2021-12-02T15:14:16",
                    "flow": {
                        "actions": [
                            {
                                "action_type": "output",
                                "port": 4
                            }
                        ],
                        "match": {
                            "dl_vlan": 300,
                            "in_port": 1
                        },
                        "priority": 10
                    }
                },
                {
                    "created_at": "2021-12-02T15:14:41",
                    "flow": {
                        "match": {
                            "in_port": 1
                        },
                        "priority": 10
                    }
                }
            ]
        },
        "00:00:00:00:00:00:00:02": {
            "0": [
                {
                    "created_at": "2021-12-02T15:04:17",
                    "flow": {
                        "actions": [
                            {
                                "action_type": "output",
                                "port": 4294967293
                            }
                        ],
                        "match": {
                            "dl_type": 35020,
                            "dl_vlan": 3799
                        },
                        "priority": 1000,
                        "table_id": 0
                    }
                }
            ]
        },
        "00:00:00:00:00:00:00:03": {
            "0": [
                {
                    "created_at": "2021-12-02T15:04:17",
                    "flow": {
                        "actions": [
                            {
                                "action_type": "output",
                                "port": 4294967293
                            }
                        ],
                        "match": {
                            "dl_type": 35020,
                            "dl_vlan": 3799
                        },
                        "priority": 1000,
                        "table_id": 0
                    }
                }
            ]
        }
    }
}```